packages/components/src/Menu/Menu.tsx

```diff
@@ -64,35 +63,46 @@ export interface SectionProps {
 export function Menu({ activator, items }: MenuProps) {
   const [visible, setVisible] = useState(false);
   const fullWidth = activator?.props?.fullWidth || false;
-  const [position, setPosition] = useState<Position>({
-    vertical: "below",
-    horizontal: "right",
-  });
-  const wrapperRef = createRef<HTMLDivElement>();
+  const shadowRef = useRef<HTMLSpanElement>(null);
+  const menuRef = useFocusTrap<HTMLDivElement>(visible);
+
+  const { width } = useWindowDimensions();
+
   const buttonID = useId();
   const menuID = useId();
 
   useOnKeyDown(handleKeyboardShortcut, ["Escape"]);
-  useSafeLayoutEffect(() => {
-    if (wrapperRef.current) {
-      const bounds = wrapperRef.current.getBoundingClientRect();
-      const newPosition = { ...position };
-
-      if (bounds.top <= window.innerHeight / 2) {
-        newPosition.vertical = "below";
-      } else {
-        newPosition.vertical = "above";
-      }
-
-      if (bounds.left <= window.innerWidth / 2) {
-        newPosition.horizontal = "right";
-      } else {
-        newPosition.horizontal = "left";
-      }
-
-      setPosition(newPosition);
-    }
-  }, [visible, fullWidth]);
+  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
+  const { styles: popperStyles, attributes } = usePopper(
+    shadowRef.current?.nextElementSibling,
+    popperElement,
+    {
+      placement: "bottom-start",
+      modifiers: [
+        {
+          name: "flip",
+          options: {
+            // fallbackPlacements: ["top-start"]
+            flipVariations: true,
+          },
+        },
+        {
+          name: "offset",
+          options: {
+            offset: [0, 0],
+          },
+        },
+      ],
+    },
+  );
+  const positionAttributes =
+    width > 489
+      ? {
+          ...attributes.popper,
+          style: popperStyles.popper,
+        }
+      : {};
```

Comment:
this is a little awkward

we're now using Popper to position the menu when it's a medium+ screen

but when it's a small screen, we don't want to use any popper logic at all. we know it should simply be position fixed at the bottom of the screen, spanning the full width. to that end, we don't want to apply any popper styles or attributes in the small screen case

I don't love that `489` is a magic number, and there isn't a super great way to share it with the media query variable value

it's also kind of meh that the position is handled by JS + CSS for biggers screens and then just CSS for the small screens